### PR TITLE
[staging-17.0] Update kmc to 17.0.318-beta

### PR DESCRIPTION
**Pull request by darcywong00**
Updating kmc compiled to current beta which required the following changes
* Add fonts to .kps files (I cherrypicked 74971f28c76aa802a6a25a5a50ffee8e32a920f2 and 3e9186595bf1e58355a505003a0962593b4876d2)
* Add definitions in the release/packages/fv_all/build.sh script

### CODE REVIEW

File: release/packages/fv_all/build.sh [hunk 1]
```diff
@@ -28,6 +28,21 @@ builder_describe_outputs \
...
+  export KMC="${REPO_ROOT}/node_modules/.bin/kmc"
+  readonly KMC
+fi
+
+# TODO: remove -W
+if [[ -z ${KMC_BUILD_PARAMS+x} ]]; then
+  export KMC_BUILD_PARAMS="build $builder_debug -W --for-publishing"
+  readonly KMC_BUILD_PARAMS
+fi
+
```
Comment (darcywong00) [inline release/packages/fv_all/build.sh:45]: I copied this block from the root build.sh

File: release/b/burushaski_girminas/source/burushaski_girminas.kps [hunk 2]
```diff
@@ -56,6 +56,18 @@
...
+      <Description>Font DejaVu Sans</Description>
+      <CopyLocation>0</CopyLocation>
+      <FileType>.ttf</FileType>
+    </File>
+    <File>
+      <Name>..\..\..\shared\fonts\dejavu\DejaVuSerif.ttf</Name>
+      <Description>Font DejaVu Serif</Description>
+      <CopyLocation>0</CopyLocation>
+      <FileType>.ttf</FileType>
+    </File>
```
Comment (mcdurdin) [inline release/b/burushaski_girminas/source/burushaski_girminas.kps:70]: Not sure why these font changes (and the next changes) are in this PR? They shouldn't impact the build, but should be in a separate PR

File: release/packages/fv_all/build.sh [hunk 3]
```diff
@@ -28,6 +28,21 @@ builder_describe_outputs \
...
+#------------------------------------------------------------
+# Definitions
+#------------------------------------------------------------
+
+if [[ -z ${KMC+x} ]]; then
+  export KMC="${REPO_ROOT}/node_modules/.bin/kmc"
+  readonly KMC
+fi
+
+# TODO: remove -W
```
Comment (mcdurdin) [inline release/packages/fv_all/build.sh:40]: We may need a follow-up PR for this

**Review by mcdurdin — CHANGES_REQUESTED** [review]
Just burushaski_girminas.kps changes that don't seem quite right

File: release/b/burushaski_girminas/source/burushaski_girminas.kps (hunk 2 above)
Comment (darcywong00) [reply release/b/burushaski_girminas/source/burushaski_girminas.kps:70]: If the font files are removed from kps, kmc fails with
```
burushaski_girminas.kpj - info KM05002: Building release/b/burushaski_girminas/burushaski_girminas.kpj
burushaski_girminas.kpj - error KM09002: File DejaVuSerif.ttf does not exist.
burushaski_girminas.kpj - info KM05007: release/b/burushaski_girminas/burushaski_girminas.kpj failed to build.
burushaski_girminas.kpj - info KM0500C: Project release/b/burushaski_girminas/burushaski_girminas.kpj failed to build.
```

Comment (mcdurdin) [reply release/b/burushaski_girminas/source/burushaski_girminas.kps:70]: You'd also need to remove the OSKFont and DisplayFont entries on ll.83-84 below, which were also added in this PR?

Comment (darcywong00) [reply release/b/burushaski_girminas/source/burushaski_girminas.kps:70]: ic - they were on the cherry-pick off of master which has the fonts
https://github.com/keymanapp/keyboards/blob/9a7accf0b1a4d25d41b7ff94c812cf1083a2b016/release/b/burushaski_girminas/source/burushaski_girminas.kps#L85-L87

I'll remove from this PR then

Comment (darcywong00) [reply release/b/burushaski_girminas/source/burushaski_girminas.kps:70]: But if I remove the ttf's from el_dinka.kps below, I get
```
el_dinka.kps - info KM05006: release/el/el_dinka/source/el_dinka.kps built successfully.
el_dinka.kpj - info KM05002: Building release/el/el_dinka/el_dinka.kpj
el_dinka.kpj - error KM09002: File DejaVuSans.ttf does not exist.
el_dinka.kpj - info KM05007: release/el/el_dinka/el_dinka.kpj failed to build.
```

Maybe due to this?
```xml
<WebDisplayFonts><Font Filename="DejaVuSans.ttf"/></WebDisplayFonts>     
```

Comment (mcdurdin) [reply release/b/burushaski_girminas/source/burushaski_girminas.kps]: el_dinka.kps: correct. The compiler now verifies the existence of the fonts in WebDisplayFonts and WebOSKFonts, which it didn't earlier in the alpha.

Comment (mcdurdin) [reply release/b/burushaski_girminas/source/burushaski_girminas.kps]: > I'll remove from this PR then

Make sure you add them back in to the next PR #2734!

**Review by mcdurdin — APPROVED** [review]
LGTM

**Pull request merged**